x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/JobModelSnapshotUpgrader.java

```diff
@@ -153,6 +159,55 @@ synchronized void start() {
         executor.execute();
     }
 
+    private void removeDuplicateModelSnapshotDoc(Consumer<Exception> runAfter) {
+        String snapshotDocId = jobId + "_model_snapshot_" + snapshotId;
+        client.prepareSearch(AnomalyDetectorsIndex.jobResultsIndexPattern())
+            .setQuery(QueryBuilders.constantScoreQuery(QueryBuilders.idsQuery().addIds(snapshotDocId)))
+            .setSize(2)
```

Comment:
Could there be more than 2 documents in this search (and, consequently, more than 1 older doc to delete)?

Reply:
@valeriy42 is there a risk of this?

Reply:
Great, thanks for explaining.
I was afraid of some potential leftover documents from previous updates. But I now see it shouldn't happen.